service/matching/tasklist/task_list_manager.go

```diff
@@ -744,11 +739,11 @@ func (c *taskListManagerImpl) DescribeTaskList(includeTaskListStatus bool) *type
 	response.TaskListStatus = &types.TaskListStatus{
 		ReadLevel:        c.taskAckManager.GetReadLevel(),
 		AckLevel:         c.taskAckManager.GetAckLevel(),
-		BacklogCountHint: backlogCount,
+		BacklogCountHint: c.taskAckManager.GetBacklogCount(),
```

Comment:
With this change `BacklogCountHint` will be capped at task pump's batch size. That may be ok. If all partitions backlog > batch_size then weighted load balancer will assign equal weights.

Reply:
Synced offline. Looks like WLB is already using ack manager's backlog count and it is performing fine. We can optimize further if needed by fetching DB backlog size in a background thread more often and incorporating it into `c.taskAckManager.GetBacklogCount()` response.